src/portal/lib/src/image-name-input/image-name-input.component.ts

```diff
@@ -16,16 +16,29 @@ export class ImageNameInputComponent implements OnInit, OnDestroy {
     selectedProjectList: Project[] = [];
     proNameChecker: Subject<string> = new Subject<string>();
     imageNameForm: FormGroup;
+    public project: string;
+    public repo: string;
+    public tag: string;
 
     constructor(
         private fb: FormBuilder,
         private errorHandler: ErrorHandler,
         private proService: ProjectService,
     ) {
         this.imageNameForm = this.fb.group({
-            projectName: ["", Validators.required],
-            repoName: ["", Validators.required],
-            tagName: ["", Validators.required],
+            projectName: ["", Validators.compose([
+                Validators.minLength(2),
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            repoName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            tagName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-zA-Z0-9_.-]+$')
```

Comment:
This RE has some problems, it should be `^[\w][\w.-]{0,127}$` (Golang).

Reply:
done.